### PR TITLE
HTML: test fieldset percentage block size

### PR DESCRIPTION
**Pull request by zcorpan**
See https://github.com/whatwg/html/pull/3934

### CODE REVIEW

**Comment by zcorpan** [comment]
Bugs:
https://bugs.webkit.org/show_bug.cgi?id=189473
https://bugzilla.mozilla.org/show_bug.cgi?id=1485646

**Comment by mstensho** [comment]
This is blocked on whatwg/html#4014 - we first need to agree on the box model here.

**Comment by zcorpan** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1485646 was closed as invalid. Need to look at this again.

**Comment by zcorpan** [comment]
@mstensho PTAL

File: html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-percentage-block-size.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ </fieldset>
+</div>
+
+<script>
+ for (const div of document.querySelectorAll('div[style]')) {
+   for (const el of div.firstElementChild.children) {
+     test(() => {
+       const expected = el.textContent === 'rendered legend' ? '80px' : '30px';
+       // 30px because: 100px - (legend that spills outside the fieldset border upwards)
+       //                     - (legend that spills outside the fieldset border downwards)
```
Comment (mstensho) [inline html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/fieldset-percentage-block-size.html]: Or 100px - (max(0, legend-block-size - border-block-start))

**Review by mstensho — APPROVED** [review]
Woho! Already passes in my LayoutNG implementation (not submitted yet). :)

**Pull request merged**